front-end/src/app/services/app-service.service.spec.ts

```diff
@@ -1,20 +1,33 @@
 import { TestBed } from '@angular/core/testing';
-
+import { HttpClientTestingModule }
+        from '@angular/common/http/testing';
 import { AppServiceService } from './app-service.service';
+import { HttpClient } from '@angular/common/http';
 
 describe('AppServiceService', () => {
   let service: AppServiceService;
+  let httpClient: HttpClient;
 
   beforeEach(() => {
-    TestBed.configureTestingModule({});
+    TestBed.configureTestingModule({
+      imports: [ HttpClientTestingModule ]
+    });
+    httpClient = TestBed.inject(HttpClient);
     service = TestBed.inject(AppServiceService);
   });
 
   it('should be created', () => {
     expect(service).toBeTruthy();
   });
+
+  it('test HttpClient getData', () => {
```

Comment:
add async here it's better, not sure it's mandatory tho
`it('test HttpClient getData', async () => {`